app/routes/task_routes.py

```diff
@@ -1 +1,104 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, Response, request
+import requests
+from app.models.task import Task
+from datetime import datetime
+from sqlalchemy import asc, desc
+from .route_utilities import validate_model, create_model, get_models_with_filters
+from ..db import db
+import os
+
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+@tasks_bp.post("")
+def create_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body:
+        return {"details": "Invalid data"}, 400
+    
+    task_data = {"title": request_body.get("title"), 
+                 "description": request_body.get("description"), 
+                 "completed_at": request_body.get("completed_at")}
+
+    task_dict, status_code = create_model(Task, task_data)
+    return {"task": task_dict}, status_code
+
+@tasks_bp.get("")
+def get_task():
+  
+    sort_order = request.args.get("sort")
+    
+    query = db.select(Task)
+    if sort_order == "asc":
+        query = query.order_by(Task.title.asc())
+    elif sort_order == "desc":
+        query = query.order_by(Task.title.desc())
+
+    tasks = db.session.scalars(query)
+    tasks_response = [task.to_dict() for task in tasks]
+
+    return tasks_response, 200
+
+@tasks_bp.get("/<task_id>")
+def get_one_task(task_id):
+    task = validate_model(Task, task_id)
+    response = {"task": task.to_dict()}
+
+    if task.goal_id is not None: 
+        response["task"]["goal_id"] = task.goal_id
```

Comment:
I would put this as `None` that way there isn't confusion where people might think you are adding a `goal_id` even though there isn't one.